Assets/MixedRealityToolkit/_Core/Managers/MixedRealityManager.cs

```diff
@@ -0,0 +1,251 @@
+        [Tooltip("The current active configuration for the Mixed Reality project")]
+        private MixedRealityConfigurationProfile activeProfile;
+
+        /// <summary>
+        /// The public property of the Active Profile, ensuring events are raised on the change of the configuration
+        /// </summary>
+        public MixedRealityConfigurationProfile ActiveProfile
+        {
+            get { return activeProfile; }
+            set { activeProfile = value; if (ProfileUpdateEvent != null) ProfileUpdateEvent.Invoke(); }
+        }
+
+        #endregion
+
+        #region Active SDK components
+
+        /// <summary>
+        /// The Active Controllers property lists all the controllers detected by the Mixed Reality manager on startup
+        /// </summary>
+        [SerializeField]
+        [Tooltip("The collection of currently active / detected controllers")]
+        private Controller[] activeControllers;
+
+        /// <summary>
+        /// The Active Headset property maintains the Headsets/SDK detected by the Mixed Reality manager on startup
+        /// </summary>
+        [SerializeField]
+        [Tooltip("The currently active / detected Headset or SDK")]
+        private Headset activeHeadset;
+
+        #endregion
+
+        #region Mixed Reality Manager Events
+
+        /// <summary>
+        /// Delegate for the Initialize event
+        /// </summary>
+        public delegate void InitializeHandler();
+        /// <summary>
+        /// Initialize event called after all managers have been registered. Informs all managers to initialize (Awake)
+        /// </summary>
+        public event InitializeHandler InitializeEvent;
+
+        /// <summary>
+        /// Delegate for the Update event
+        /// </summary>
+        public delegate void UpdateHandler();
+        /// <summary>
+        /// The Update event delegates the MonoBehaviour update message, informing all active managers to update (which is faster as managers are not bound to a MonoBehaviour)
+        /// </summary>
+        public event UpdateHandler UpdateEvent;
+
+        /// <summary>
+        /// Delegate for the Destroy event
+        /// </summary>
+        public delegate void DestroyHandler();
+        /// <summary>
+        /// The Destroy event is fired before the Mixed Reality Manager is destroyed in a scene and informs all active managers to clean up prior to close. Linked to the MonoBehaviour Destroy event.
+        /// </summary>
+        public event DestroyHandler DestroyEvent;
+
+        /// <summary>
+        /// Delegate for the Profile Update event
+        /// </summary>
+        public delegate void ProfileUpdateHandler();
+        /// <summary>
+        /// The ProfileUpdate event fires whenever the MixedRealityManager active profile is replaced.
+        /// *Note Profile options cannot be changed once the project is running.
+        /// </summary>
+        public event ProfileUpdateHandler ProfileUpdateEvent;
```

Comment:
Instead of ProfileUpdate, let's call this "Reset" so the managers can do what they need.  Also gives us a more generic way to reset the managers as needed.